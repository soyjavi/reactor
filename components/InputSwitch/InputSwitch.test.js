import React from 'react';
import renderer from 'react-test-renderer';

import { InputSwitch } from '..';

describe('<InputSwitch>', () => {
  it('renders', () => {
    const tree = renderer.create(<InputSwitch />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {color}', () => {
    const tree = renderer.create(<InputSwitch color="green" value />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {label}', () => {
    const tree = renderer.create(<InputSwitch label="Hello world" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {disabled}', () => {
    const tree = renderer.create(<InputSwitch visible />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<InputSwitch style={{ backgroundColor: 'orange' }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
