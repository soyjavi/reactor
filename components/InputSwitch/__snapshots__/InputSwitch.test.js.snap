// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputSwitch> renders 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
        "height": "auto",
        "width": "100%",
      },
      Object {
        "alignItems": "center",
      },
      Object {
        "justifyContent": "flex-start",
      },
      undefined,
      Object {
        "height": 40,
      },
    ]
  }
>
  <RCTSwitch
    accessibilityRole="switch"
    activeThumbColor="#ffffff"
    activeTrackColor="#191919"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "height": 24,
          "marginRight": 8,
        },
      ]
    }
    thumbTintColor="#ffffff"
    value={false}
  />
</View>
`;

exports[`<InputSwitch> when {color} 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
        "height": "auto",
        "width": "100%",
      },
      Object {
        "alignItems": "center",
      },
      Object {
        "justifyContent": "flex-start",
      },
      undefined,
      Object {
        "height": 40,
      },
    ]
  }
>
  <RCTSwitch
    accessibilityRole="switch"
    activeThumbColor="#ffffff"
    activeTrackColor="green"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "height": 24,
          "marginRight": 8,
        },
      ]
    }
    thumbTintColor="#ffffff"
    value={true}
  />
</View>
`;

exports[`<InputSwitch> when {disabled} 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
        "height": "auto",
        "width": "100%",
      },
      Object {
        "alignItems": "center",
      },
      Object {
        "justifyContent": "flex-start",
      },
      undefined,
      Object {
        "height": 40,
      },
    ]
  }
>
  <RCTSwitch
    accessibilityRole="switch"
    activeThumbColor="#ffffff"
    activeTrackColor="#191919"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "height": 24,
          "marginRight": 8,
        },
      ]
    }
    thumbTintColor="#ffffff"
    value={false}
  />
</View>
`;

exports[`<InputSwitch> when {label} 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
        "height": "auto",
        "width": "100%",
      },
      Object {
        "alignItems": "center",
      },
      Object {
        "justifyContent": "flex-start",
      },
      undefined,
      Object {
        "height": 40,
      },
    ]
  }
>
  <RCTSwitch
    accessibilityRole="switch"
    activeThumbColor="#ffffff"
    activeTrackColor="#191919"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "height": 24,
          "marginRight": 8,
        },
      ]
    }
    thumbTintColor="#ffffff"
    value={false}
  />
</View>
`;

exports[`<InputSwitch> when {style} 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
        "height": "auto",
        "width": "100%",
      },
      Object {
        "alignItems": "center",
      },
      Object {
        "justifyContent": "flex-start",
      },
      undefined,
      Object {
        "height": 40,
      },
    ]
  }
>
  <RCTSwitch
    accessibilityRole="switch"
    activeThumbColor="#ffffff"
    activeTrackColor="#191919"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "height": 24,
          "marginRight": 8,
        },
      ]
    }
    thumbTintColor="#ffffff"
    value={false}
  />
</View>
`;
